Sources/SpeziAccessGuard/CodeViews/EnterCodeView.swift

```diff
@@ -23,14 +24,24 @@ struct EnterCodeView: View {
                     Text("ACCESS_CODE_PASSCODE_PROMPT", bundle: .module)
                         .font(.title2)
                         .frame(maxWidth: .infinity)
-                    CodeView(codeOption: codeOption) { code in
-                        do {
-                            try await viewModel.checkAccessCode(code)
-                        } catch {
-                            wrongCodeCounter += 1
-                            let errorMessageTemplate = NSLocalizedString("ACCESS_CODE_PASSCODE_ERROR %@", bundle: .module, comment: "")
-                            errorMessage = String(format: errorMessageTemplate, "\(wrongCodeCounter)")
-                            throw error
+                    CodeView(codeOption: codeOption,
+                             toolbarButtonLabel: String(localized: "ENTER_PASSCODE_CONFIRM_BUTTON", bundle: .module)) { code, validationRes in
+                        
+                        switch validationRes {
+                            case .valid:
+                                errorMessage = nil
+                            case .failure(let upstreamError):
+                                errorMessage = String(upstreamError.failureReason)
```

Comment:
Casting might not be needed here:
```suggestion
                                errorMessage = upstreamError.failureReason
```

Reply:
I know, I thought the same, but for some reason it is required. I will check again and find out whats going wrong.

Reply:
I did find a workaround with a second property `.errorMessage` (see next commit). However, wasn't able to achieve this without explicit casting with the `.failureReason` directly.